integration-tests/src/test/java/org/apache/druid/tests/query/ITSqlQueryTest.java

```diff
@@ -55,19 +55,23 @@ public class ITSqlQueryTest
   @Inject
   IntegrationTestingConfig config;
 
+  @Inject
+  @TestClient
+  HttpClient httpClient;
+
   interface IExecutable
   {
-    void execute(String endpoint) throws IOException;
+    void execute(String endpoint) throws Exception;
   }
 
   interface OnRequest
   {
-    void on(HttpPost request) throws IOException;
+    void on(Request request) throws IOException;
   }
 
   interface OnResponse
   {
-    void on(int statusCode, HttpEntity response) throws IOException;
+    void on(int statusCode, String response) throws IOException;
   }
 
   private void executeWithRetry(String endpoint, String contentType, IExecutable executable)
```

Comment:
Maybe use `ITRetryUtils` in this method.

Reply:
resolved